src/window/mouse_manager.rs

```diff
@@ -69,40 +74,76 @@ struct TouchTrace {
 }
 
 pub struct MouseManager {
-    dragging: Option<String>,
+    drag_details: Option<DragDetails>,
     drag_position: PhysicalPosition<u32>,
 
     has_moved: bool,
-    position: PhysicalPosition<u32>,
-    relative_position: PhysicalPosition<u32>,
+    position: PhysicalPosition<f32>,
 
     scroll_position: PhysicalPosition<f32>,
 
     // the tuple allows to keep track of different fingers per device
     touch_position: HashMap<(DeviceId, u64), TouchTrace>,
 
-    window_details_under_mouse: Option<WindowDrawDetails>,
-
     mouse_hidden: bool,
     pub enabled: bool,
 }
 
 impl MouseManager {
     pub fn new() -> MouseManager {
         MouseManager {
-            dragging: None,
+            drag_details: None,
             has_moved: false,
-            position: PhysicalPosition::new(0, 0),
-            relative_position: PhysicalPosition::new(0, 0),
+            position: PhysicalPosition::new(0.0, 0.0),
             drag_position: PhysicalPosition::new(0, 0),
             scroll_position: PhysicalPosition::new(0.0, 0.0),
             touch_position: HashMap::new(),
-            window_details_under_mouse: None,
             mouse_hidden: false,
             enabled: true,
         }
     }
 
+    fn get_window_details_under_mouse<'a>(
+        &self,
+        renderer: &'a Renderer,
+    ) -> Option<&'a WindowDrawDetails> {
+        let position = self.position;
+
+        // the rendered window regions are sorted by draw order, so the earlier windows in the
+        // list are drawn under the later ones
+        renderer
+            .window_regions
+            .iter()
+            .filter(|details| {
+                position.x >= details.region.left
+                    && position.x < details.region.right
+                    && position.y >= details.region.top
+                    && position.y < details.region.bottom
+            })
+            .last()
+    }
+
+    fn get_relative_position(
```

Comment:
Minor thing, but would it make sense to put this as a function on the renderer which takes a position and returns a relative position?

Take it or leave it

Reply:
I think it's slightly too specialized, since it takes a global position in pixels, then clamps the coordinates to an active Neovim window, and returns the relative grid coordinates. So, I left it as it is.